Assets/Scripts/Gamestate/MatchController.cs

```diff
@@ -0,0 +1,141 @@
+{
+    public MatchPlayer(Player player, PlayerStateController playerStateController, int startAmount)
+    {
+        this.player = player;
+        playerStateController.chips = startAmount;
+        this.playerStateController = playerStateController;
+    }
+    // Reference to player identity class
+    public Player player;
+    // Reference to in-match player
+    public PlayerStateController playerStateController;
+}
+
+public class MatchController : MonoBehaviour
+{
+    public static MatchController Singleton { get; private set; }
+
+    public delegate void MatchEvent();
+
+    public MatchEvent onRoundEnd;
+    public MatchEvent onRoundStart;
+    public MatchEvent onBiddingStart;
+    public MatchEvent onBiddingEnd;
+
+    [Header("Chip rewards")]
+    [SerializeField]
+    private int startAmount = 0;
+    [SerializeField]
+    private int rewardWin = 2;
+    [SerializeField]
+    private int rewardKill = 1;
+    [SerializeField]
+    private int rewardBase = 1;
+
+    private List<MatchPlayer> matchPlayers = new List<MatchPlayer>();
+    private List<Round> rounds = new List<Round>();
+
+    void Start()
+    {
+        #region Singleton boilerplate
+
+        if (Singleton != null)
+        {
+            if (Singleton != this)
+            {
+                Debug.LogWarning($"There's more than one {Singleton.GetType()} in the scene!");
+                Destroy(gameObject);
+            }
+
+            return;
+        }
+
+        Singleton = this;
+
+        #endregion Singleton boilerplate
+
+        PlayerInputManagerController.Singleton.playerInputs.ForEach(playerInput =>
+        {
+            var player = playerInput.gameObject.GetComponent<Player>();
+            var playerStateController = playerInput.gameObject.transform.parent.GetComponent<PlayerStateController>();
+            matchPlayers.Add(new MatchPlayer(player, playerStateController, startAmount));
+        });
+
+        // TODO do something else funky wunky
+        onRoundEnd += () => Debug.Log("End of round " + rounds.Count());
+        StartNextRound();
+    }
+
+    public void StartNextRound()
+    {
+        onRoundStart?.Invoke();
+        rounds.Add(new Round(matchPlayers.Select(player => player.playerStateController).ToList()));
+    }
+
+    public void EndActiveRound()
+    {
+        onRoundEnd?.Invoke();
+
+        AssignRewards();
+
+        CheckWinCondition();
+    }
+
+    private void AssignRewards()
+    {
+        foreach (MatchPlayer player in matchPlayers)
+        {
+            player.playerStateController.chips +=
+                // Base reward
+                rewardBase
+                // Kill bonus
+                + (rounds.Last().kills.ContainsKey(player.playerStateController) ?
+                rounds.Last().kills[player.playerStateController].Count * rewardKill : 0)
+                // Winner bonus
+                + (player.playerStateController == rounds.Last().winner ? rewardWin : 0);
+        }
+    }
```

Comment:
Although I also tend to (mis)use the ternary operator, I find that in this scenario, we might be better off with more expressive if-statements, or even better: methods defining these stats in the `Round`-class (see comment on Round.cs)

e.g:
```cs
// MatchController.cs
private void AssignRewards()
{
    var lastRound = round.Last();
    foreach (MatchPlayer player in matchPlayers)
    {
        var reward = rewardBase + lastRound.KillCount(player.playerStateController) * rewardKill;
        if (lastRound.IsWinner(player.playerStateController)
           reward += rewardWin;

        player.playerStateController.chips += reward;
    }
}
```